core/chains/evm/client/client.go

```diff
@@ -26,19 +27,16 @@ const queryTimeout = 10 * time.Second
 
 // Client is the interface used to interact with an ethereum node.
 type Client interface {
+	cmntypes.Client[*big.Int, common.Address, types.Block, types.Transaction, common.Hash, types.Receipt, types.Log, ethereum.FilterQuery]
+
 	Dial(ctx context.Context) error
 	Close()
-	ChainID() *big.Int
```

Comment:
As discussed in another comment thread, lets keep this function signature as it is, and return this value from configured one, and NOT call RPC.